File: src/Environment.cpp
#include "Environment.h"
#include "TicTacToeEnvironment.h"

Environment::Environment(int numObservations, int numActions)
: observationSpace(numObservations), actionSpace(numActions)
{}

Environment::~Environment()
{}

std::vector<double> Environment::reset()
{
    return observationSpace;
}

std::tuple<std::vector<double>, float, bool> Environment::step(int action)
{
        
        return std::make_tuple(observationSpace, 0.0f, false);
}

void Environment::render()
{
    return;
}

void Environment::close()
{
    return;
}

File: src/environment.py
class Environment:
    def __init__(self, action_space, observation_space):
        self.action_space = action_space
        self.observation_space = observation_space

    def reset(self):
        pass

    def step(self):
        pass

    def close(self):
        pass

    def render(self):
        pass